ledger/src/blockstore.rs

```diff
@@ -176,7 +176,7 @@ impl Blockstore {
     }
 
     /// Opens a Ledger in directory, provides "infinite" window of shreds
-    pub fn open(ledger_path: &Path) -> Result<Blockstore> {
+    pub fn open(ledger_path: &Path, access_type: AccessType) -> Result<Blockstore> {
```

Comment:
Looks like this signature change caused a ton of churn.  How about we leave `open()` signature unchanged, making it behave as AllowSecondary, then  add `open_with_access_type()` for the fewer times that the caller wants a different AcessType

Reply:
Yeah, considering back-porting, less churn is desirable. I did as you suggested except the default being `OnlyPrimary` not to change the behavior for most of existing code path. We need to restrict to use `AllowSecondary` only for read-only access like from `ledger-tool` inspection subcommands.